src/components/SearchableMultiSelect/SearchableMultiSelect.jsx

```diff
@@ -0,0 +1,397 @@
+		 * whether it should be visible or not. The default behavior of the
+		 * function is to match, ignoring case, against any text node descendant of
+		 * the `Option`.
+		 *
+		 * Signature: `(searchText, optionProps) => {}`
+		 *
+		 * If `true` is returned, the option will be visible. If `false`, the
+		 * option will not be visible.
+		 */
+		optionFilter: func,
+		/**
+		 * The current search text to filter the list of options by.
+		 */
+		searchText: string,
+		/**
+		 * An array of currently selected `SearchableMultiSelect.Option` indices or
+		 * `null` if nothing is selected.
+		 */
+		selectedIndices: arrayOf(number),
+		/**
+		 * Object of DropMenu props which are passed through to the underlying
+		 * DropMenu component.
+		 */
+		DropMenu: shape(DropMenu.propTypes),
+		/**
+		 * Object of SearchField props which are passed through to the underlying
+		 * SearchField component.
+		 */
+		SearchField: shape(SearchField.propTypes),
+		/**
+		 * *Child Element* - These are menu options. Each `Option` may be passed a
+		 * prop called `isDisabled` to disable selection of that `Option`. Any
+		 * other props pass to Option will be available from the `onSelect`
+		 * handler.
+		 *
+		 * It also support the `Selection` prop that can be used to forward along
+		 * props to the underlying `Selection` component.
+		 */
+		Option: any,
+		/**
+		 * Adjusts the display of this component. This should typically be driven
+		 * by screen size. Currently `small` and `large` are explicitly handled
+		 * by this component.
+		 */
+		responsiveMode: oneOf(['small', 'medium', 'large']),
+		/**
+		 * Controls the visibility of the "remove all" button that's shown with the
+		 * selected items.
+		 */
+		hasRemoveAll: bool,
+	},
+
+	getDefaultProps() {
+		return {
+			isDisabled: false,
+			isLoading: false,
+			onRemoveAll: _.noop,
+			optionFilter: propsSearch,
+			searchText: '',
+			selectedIndices: [],
+			DropMenu: DropMenu.getDefaultProps(),
+			SearchField: SearchField.getDefaultProps(),
+			responsiveMode: 'large',
+			hasRemoveAll: true,
+		};
+	},
+
+	handleDropMenuSelect(optionIndex, { event, props }) {
+		const { onSelect } = this.props;
+
+		return onSelect(optionIndex, { event, props });
+	},
+
+	handleCheckboxSelect(_isSelected, {
+		// TODO: make sure the consumer can do callbackId somehow
+		event,
+		props: { callbackId: optionIndex },
+	}) {
+		// This is needed otherwise clicking the checkbox will double fire this
+		// event _and_ the `handleDropMenuSelect` handler
+		event.stopPropagation();
+
+		// We don't want to send the consumer the checkbox's props so we have to
+		// lookup the option they clicked and send its props along
+		const selectedOptionProps = _.get(findTypes(this.props, SearchableMultiSelect.Option), `[${optionIndex}].props`);
+
+		return this.props.onSelect(optionIndex, { event, props: selectedOptionProps });
+	},
+
+	handleSelectionRemove({
+		event,
+		props,
```

Comment:
I was doing this for backwards compatibility with our old convention. However, you're probably right that moving forward we should just start using the new convention.